src/util/expandThoughts.js

```diff
@@ -80,31 +82,36 @@ export const expandThoughts = (path, thoughtIndex, contextIndex, contextViews =
   return (isOnlyChildNoUrl || isTable() || pinChildren() || publishPinChildren
     ? children
     : children.filter(child => {
+      const childPath = (path || []).concat(child)
+      const isCursor = equalPath(childPath, cursor)
```

Comment:
Since only `child` changes on each iteration, you only need to compare `child` with `head(cursor)`.